agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4251,7 +4254,67 @@ bool slave_thread::handle_client_capability_query(Socket *sd, ieee1905_1::CmduMe
 {
     const auto mid = cmdu_rx.getMessageId();
     LOG(DEBUG) << "Received CLIENT_CAPABILITY_QUERY_MESSAGE , mid=" << std::dec << int(mid);
-    return true;
+
+    auto client_info_tlv_r = cmdu_rx.getClass<wfa_map::tlvClientInfo>();
+    if (!client_info_tlv_r) {
+        LOG(ERROR) << "getClass wfa_map::tlvClientInfo failed";
+        return false;
+    }
+
+    //Check if it is an error scenario - if the STA specified in the Client Capability Query message is not associated
+    //with any of the BSS operated by the Multi-AP Agent (an error scenario)
+    bool error_scenario = 1;
+    auto it             = m_vaps_clients_map.find(client_info_tlv_r->bssid());
+    if (it != m_vaps_clients_map.end()) {
+        if (it->second.count(client_info_tlv_r->client_mac())) {
+            //only if the client exist in one of the BSS operated by the Multi-AP Agent this is not an error scenario
+            error_scenario = 0;
+        }
+    }
+
+    // send CLIENT_CAPABILITY_REPORT_MESSAGE back to the controller
+    if (!cmdu_tx.create(mid, ieee1905_1::eMessageType::CLIENT_CAPABILITY_REPORT_MESSAGE)) {
+        LOG(ERROR) << "cmdu creation of type CLIENT_CAPABILITY_REPORT_MESSAGE, has failed";
+        return false;
+    }
+
+    auto client_info_tlv_t = cmdu_tx.addClass<wfa_map::tlvClientInfo>();
+    if (!client_info_tlv_t) {
+        LOG(ERROR) << "addClass wfa_map::tlvClientInfo has failed";
+        return false;
+    }
+    client_info_tlv_t->bssid()      = client_info_tlv_r->bssid();
```

Comment:
bssid may have changed already...

For all we know, stupid controllers may just leave it at 00:00:00:00:00:00.